corehq/apps/accounting/invoicing.py

```diff
@@ -353,6 +326,163 @@ def create_wire_credits_invoice(self, items, amount):
         )
 
 
+class CustomerAccountInvoiceFactory(object):
+    """
+        This generates an invoice for a Customer Billing Account.
+    """
+    def __init__(self, date_start, date_end, account, recipients=None):
+        """
+        The Invoice generated will always be for the month preceding the
+        invoicing_date.
+        For example, if today is July 5, 2014 then the invoice will be from
+        June 1, 2014 to June 30, 2014.
+        """
+        self.date_start = date_start
+        self.date_end = date_end
+        self.account = account
+        self.recipients = recipients
+        self.customer_invoice = None
+
+    def create_invoice(self):
+        invoices = []
+        for subscription in self.account.subscription_set.all():
+            if should_create_invoice(subscription, subscription.subscriber.domain):
+                try:
+                    invoice_start = self._get_invoice_start(subscription, self.date_start)
+                    invoice_end = self._get_invoice_end(subscription, self.date_end)
+                    with transaction.atomic():
+                        invoices.append(self._generate_invoice(subscription, invoice_start, invoice_end))
+                except InvoiceAlreadyCreatedError as e:
+                    log_accounting_error(
+                        "Invoice already existed for domain %s: %s." % (subscription.subscriber.domain, e),
+                        show_stack_trace=True
+                    )
+        self.customer_invoice = self._consolidate_invoices(invoices)
+        self._email_invoice()
+
+    def _generate_invoice(self, subscription, invoice_start, invoice_end):
```

Comment:
From a quick glance this function looks identical to ```DomainInvoiceFactory._generate_invoice```.  Is it possible to refactor this to avoid code duplication?